joomla/joomla-cms
# [4] Move JS file to the build folder

Pull Request for Issue # .

### Summary of Changes
- Change the move to top button to use Intersection observer instead onScroll event (perf)
- move the .js to the build folder and convert it to ES20xx
- use type=module for the script

IntersectionObserver Refs:
- MDN: https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API 
- CanIUse: https://caniuse.com/intersectionobserver

### Testing Instructions
Fetch the PR
run `npm install`
Check that the Cassiopeia move to the top button still works as expected

### Actual result BEFORE applying this Pull Request


### Expected result AFTER applying this Pull Request



### Documentation Changes Required

No this is just a rewrite of the script with modern JS, the functionality is the same

@wilsonge the js file although it's in the media folder **IS NOT OVERRIDDABLE** but it's just a step towards converting the Cassiopeia to parent template (eg supporting child templates)
